client/state/happychat/test/selectors.js

```diff
@@ -58,6 +70,59 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( '#canUserSendMessages', () => {
+		const messagingDisabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ABANDONED,
+			HAPPYCHAT_CHAT_STATUS_BLOCKED,
+			HAPPYCHAT_CHAT_STATUS_DEFAULT,
+			HAPPYCHAT_CHAT_STATUS_MISSED,
+			HAPPYCHAT_CHAT_STATUS_PENDING,
+		];
+		const messagingEnabledChatStatuses = [
+			HAPPYCHAT_CHAT_STATUS_ASSIGNED,
+			HAPPYCHAT_CHAT_STATUS_ASSIGNING,
+			HAPPYCHAT_CHAT_STATUS_CLOSED,
+			HAPPYCHAT_CHAT_STATUS_NEW,
+		];
+
+		it( 'should return false if Happychat is unavailable', () => {
+			const state = {
+				happychat: {
+					connectionStatus: 'uninitialized',
```

Comment:
We don't — I didn't want to make the test cases for this PR larger by adding those and modifying a bunch of unrelated stuff. This should be addressed though.